Support for additional Abode timeline events

libots opened the pull request:
Adds support for additional Abode timeline events to fire an event in HA:
- **abode_disarm**: Fired when the alarm is disarmed.
- **abode_arm**: Fired when the alarm is armed (home or away).
- **abode_test**: Fired when a sensor is in test mode.
- **abode_capture**: Fired when an image is captured.
- **abode_device**: Fired for device changes/additions/deletions.
- **abode_automation_edit**: Fired for changes to automations.

Additionally, exposes **app_type** and **event_by** fields from the Abode timeline, which can be used to filter for events initiated by logged in users (i.e., Abode web app, iOS app, AbodePy/HA, etc.) as well as keypad users (new feature).

**Pull request with documentation for [home-assistant.io](https://github.com/home-assistant/home-assistant.io) (if applicable):** home-assistant/home-assistant.io#10956

## Checklist:
  - [x] The code change is tested and works locally.
  - [x] Local tests pass with `tox`. **Your PR cannot be merged unless tests pass**
  - [x] There is no commented out code in this PR.
  - [x] I have followed the [development checklist][dev-checklist]

If user exposed functionality or configuration variables are added/changed:
  - [x] Documentation added/updated in [home-assistant.io](https://github.com/home-assistant/home-assistant.io)

If the code does not interact with devices:
  - [x] Tests have been added to verify that the new code works.

[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html

libots committed:
Support for additional timeline events

libots committed:
Update __init__.py

balloob commented on homeassistant/components/abode/__init__.py:
```diff
@@ -259,6 +259,12 @@ def event_callback(event, event_json):
         TIMELINE.PANEL_FAULT_GROUP,
         TIMELINE.PANEL_RESTORE_GROUP,
         TIMELINE.AUTOMATION_GROUP,
+        TIMELINE.DISARM_GROUP,
```
We shouldn't add events that can be derived from the state of the alarm control panel entity changing.

libots replied:
The events provide more information than just the state change, notably the user who accomplished the action. I can see a few places where this would be useful, such as doing different things if children, landlords, contractors, cleaners, etc. disarm the system.

libots committed:
Expose details on user

These lines expose apptype and event_by, which can be used to give information on events initiated by keypad users (vs. users on the mobile app, web app, or those initiated from HA through abodepy).
